client/lib/user/user.d.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * External Dependencies
+ */
+import { EventEmitter } from 'events';
+
+export class User extends EventEmitter {
+	initialize: () => Promise< void >;
+	clearStoreIfChanged: () => void;
```

Comment:
did you check these on their return value? I think TypeScript won't like it if we specify `void` as return when we mean "I don't care"

alternatives here might be `any` which definitely won't break and also `unknown` in case we want to try and force people not to use them.

sometimes the function returns a value and we just don't care; other times someone might be using an early-abort with a debug call…

```ts
if (someCondition) {
    return debug('Some condition met');
}
```

…and there's no need to consider that a type error if it doesn't cause problems at the called function's definition

Reply:
I went through these and checked that the function signatures match what's actually being returned. Quite a few of them were incorrect, oops. I made the type signature return void if the method only returns `undefined` or doesn't have any return statements. Does that sound correct?

Reply:
> made the type signature return void if the method only returns undefined or doesn't have any return statements. Does that sound correct?

That sounds correct!